addons/mail/models/ir_attachment.py

```diff
@@ -71,5 +82,98 @@ def _attachment_format(self, commands=False):
+        If the URL is an image, this will create an image attachment.
+        """
+        try:
+            page = requests.get(url, timeout=1)
+        except requests.exceptions.RequestException:
+            return False
+
+        if page.status_code != requests.codes.ok:
+            return False
+
+        image_mimetype = [
+            'image/bmp',
+            'image/gif',
+            'image/jpeg',
+            'image/png',
+            'image/tiff',
+            'image/x-icon',
+        ]
+        if page.headers['Content-Type'] in image_mimetype:
+            return self._get_data_from_url_image(url)
+        elif 'text/html' in page.headers['Content-Type']:
+            return self._get_data_from_url_html(url, page.content)
+        return False
+
+    def _get_data_from_url_image(self, url):
+        image = self._get_image_from_url(url)
+        data = {
+            'url': url,
+            'name': url,
+            'description': False,
+            'raw': image,
+            'mimetype': 'image/o-linkpreview-image',
+        }
+        return data
+
+    def _get_image_from_url(self, image_url):
+        request_image = False
+        image = False
+        try:
+            request_image = requests.get(image_url, timeout=1)
+        except requests.exceptions.RequestException:
+            request_image = False
+        if request_image and request_image.status_code != requests.codes.ok:
+            return False
+        if (request_image):
+            image = image_process(
+                request_image.content,
+                size=(300, 300),
+                verify_resolution=True
+            )
+        return image
+
+    def _get_data_from_url_html(self, url, content):
+        tree = html.fromstring(content)
+        title = tree.xpath('//meta[@property="og:title"]/@content')
+        if title:
+            image_url = tree.xpath('//meta[@property="og:image"]/@content')
+            image = False
+            if image_url:
+                image = self._get_image_from_url(image_url[0])
+            description = tree.xpath('//meta[@property="og:description"]/@content')
+            data = {
+                'url': url,
+                'name': title[0] if title else url,
+                'raw': image if image else False,
+                'description': description[0] if description else False,
+                'mimetype': 'application/o-linkpreview-with-thumbnail' if image else 'application/o-linkpreview',
+            }
+            return data
+        return False
+
+    @api.autovacuum
+    def _gc_link_preview(self):
+        mimetypes = [
+            'application/o-linkpreview',
+            'image/o-linkpreview-image',
+            'application/o-linkpreview-with-thumbnail',
+        ]
+        date_to_delete = fields.Date.to_string((datetime.now() + relativedelta(days=-7)))
+        domain = [
+            ('mimetype', 'in', mimetypes),
+            ('write_date', '<', date_to_delete)
```

Comment:
Missing trailing comma.
```suggestion
            ('write_date', '<', date_to_delete),
```